src/printer.js

```diff
@@ -367,16 +367,28 @@ function printExpression(path, options, print) {
           "::",
           path.call(print, "offset")
         ]);
-      case "offsetlookup":
+      case "offsetlookup": {
+        const isOffsetNumberNode = node.offset && node.offset.kind === "number";
+
         return group(
           concat([
             path.call(print, "what"),
             "[",
-            group(indent(concat([softline, path.call(print, "offset")]))),
-            softline,
+            group(
+              concat([
```

Comment:
is the extra ` concat()` necessary?

Reply:
@mgrip need, for `isOffsetNumberNode ? "" : softline`

Reply:
👍